src/proto/Solana.proto

```diff
@@ -0,0 +1,44 @@
+syntax = "proto3";
+
+package TW.Solana.Proto;
+option java_package = "wallet.core.jni.proto";
+
+message Transfer {
+    bytes private_key = 1;
+    string recipient = 2;
+    uint64 value = 3;
+}
+
+message Stake {
+    bytes from_private_key = 1;
+    bytes stake_private_key = 2;
```

Comment:
can we just use one private key? it would be the same

Reply:
The solana stake transaction contains 2 different instructions:
- Create and fund stake account (allocate space) at stake_private_key corresponding address; needs to be signed by the funder (from_private_key)
- Initialize stake account state; needs to be signed by the stake key (stake_private_key)

We batch these actions into one transaction so they are executed atomically.

They could be split apart into two transactions, but would still depend on two different private keys.

How should we handle?

Reply:
Is `Stake` meaning Delegation? because we only need delegation messages, because wallet itself won't be running staking, just delegation

Reply:
either way, user will have only one private key for solana

Reply:
Yes, delegation.
We'll need to do a little rework on the solana side on our stake program to support a single key for this.